### ISSUE TITLE
Predicate function names check does not work for non-zero arity functions

### ISSUE THREAD

**Issue by azhi** [rrrene/credo#1084]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* **_For bugs, please do a quick search and make sure the bug has not yet been reported here_**

This is a duplicate of https://github.com/rrrene/credo/issues/866, which was first [resolved](https://github.com/rrrene/credo/pull/867), but then [reverted](https://github.com/rrrene/credo/commit/9638ba524a0c3573a0ceed063e79724009173928). I would reopen initial issue if I could :).

### Environment

* Credo version (`mix credo -v`): 1.7.1-ref.develop.1c0343b3d+uncommittedchanges
* Erlang/Elixir version (`elixir -v`): Erlang 25, Elixir 1.14.1
* Operating system: ArchLinux

### What were you trying to do?

1. Have following function definition

```elixir
defmodule A do
  defp is_user(cookie) do
    true
  end
end
```

2. Credo has `PredicateFunctionNames` check enabled.
3. Run credo.

### Expected outcome

Credo reports an issue with the function - it is a predicate function that starts with `is_`.

### Actual outcome

Credo does not report any issues.

### Extra information

The check works fine if I make the function 0-arity, for the following code an issue is reported:

```elixir
defmodule A do
  defp is_user do
    true
  end
end
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by azhi** [comment]
Yep, it works now, thanks.
